src/Servers/Kestrel/Core/src/Internal/Http2/Http2OutputProducer.cs

```diff
@@ -64,7 +78,7 @@ public Http2OutputProducer(Http2Stream stream, Http2StreamContext context, Strea
 
         public void StreamReset()
         {
-            Debug.Assert(_dataWriteProcessingTask.IsCompletedSuccessfully);
+            Debug.Assert(!_dataWriteProcessingTask.IsCompletedSuccessfully);
```

Comment:
You want to assert that _responseCompleteWaitSource.GetStatus(_responseCompleteWaitSource.Version) == Completed

Reply:
Are Debug builds running as part of CI atm?

Reply:
I don't think so. It is mainly to catch errors during dev and communicate what the state of the class should be at this point.